extensions/blocks/podcast-player/edit.js

```diff
@@ -81,6 +83,59 @@ const PodcastPlayerEdit = ( {
 	const [ editedUrl, setEditedUrl ] = useState( url || '' );
 	const [ isEditing, setIsEditing ] = useState( false );
 	const [ feedData, setFeedData ] = useState( {} );
+	const cancellableFetch = useRef();
+	const [ isInteractive, setIsInteractive ] = useState( false );
+
+	const fetchFeed = useCallback(
+		urlToFetch => {
+			const encodedURL = encodeURIComponent( urlToFetch );
+
+			cancellableFetch.current = makeCancellable(
+				apiFetch( {
+					path: '/wpcom/v2/podcast-player?url=' + encodedURL,
+				} )
+			);
+
+			cancellableFetch.current.promise.then(
+				data => {
+					if ( data?.isCanceled ) {
+						debug( 'Block was unmounted during fetch', data );
+						return; // bail if canceled to avoid setting state
+					}
+					// Store feed data.
+					setFeedData( data );
+				},
+				error => {
+					if ( error?.isCanceled ) {
+						debug( 'Block was unmounted during fetch', error );
+						return; // bail if canceled to avoid setting state
+					}
+					if ( /\bspotify\b/i.test( encodedURL ) ) {
+						createErrorNotice(
+							__(
+								"It looks like you're trying to embed a podcast hosted on Spotify. Please use the Spotify block instead.",
+								'jetpack'
+							)
```

Comment:
Maybe as a follow-up, we could rely on `createBlock` and switch blocks for the user?

Reply:
Exactly! We are planning to change this into a notice with a button that automatically creates the appropriate block with the URL. As part of that, we might be adding more providers and we'll definitely need to restructure the code a bit so it's a bit separate from the normal error reporting.

Reply:
#15275